src/vmm/src/signal_handler.rs

```diff
@@ -128,162 +156,95 @@ generate_handler!(
+    SyscallReturnCode(unsafe { libc::sigemptyset(&mut mask as *mut sigset_t) })
+        .into_empty_result()
+        .map_err(Error::SyscallError)?;
 
-    // This function is used when running unit tests, so all the unsafes are safe.
-    fn cpu_count() -> usize {
-        let mut cpuset: cpu_set_t = unsafe { mem::zeroed() };
-        unsafe {
-            libc::CPU_ZERO(&mut cpuset);
-        }
-        let ret = unsafe {
-            libc::sched_getaffinity(
-                0,
-                mem::size_of::<cpu_set_t>(),
-                &mut cpuset as *mut cpu_set_t,
-            )
-        };
-        assert_eq!(ret, 0);
-
-        let mut num = 0;
-        for i in 0..libc::CPU_SETSIZE as usize {
-            if unsafe { libc::CPU_ISSET(i, &cpuset) } {
-                num += 1;
-            }
-        }
-        num
+    // Add all handled signals to the mask set.
+    for signal in HANDLED_SIGNALS.iter() {
+        SyscallReturnCode(unsafe { libc::sigaddset(&mut mask as *mut sigset_t, *signal) })
+            .into_empty_result()
+            .map_err(Error::SyscallError)?;
     }
 
-    #[test]
-    fn test_signal_handler() {
-        let child = thread::spawn(move || {
-            assert!(register_signal_handlers().is_ok());
-
-            let filter = SeccompFilter::new(
-                vec![
-                    allow_syscall(libc::SYS_brk),
-                    allow_syscall(libc::SYS_exit),
-                    allow_syscall(libc::SYS_futex),
-                    allow_syscall(libc::SYS_getpid),
-                    allow_syscall(libc::SYS_munmap),
-                    allow_syscall(libc::SYS_kill),
-                    allow_syscall(libc::SYS_rt_sigprocmask),
-                    allow_syscall(libc::SYS_rt_sigreturn),
-                    allow_syscall(libc::SYS_sched_getaffinity),
-                    allow_syscall(libc::SYS_set_tid_address),
-                    allow_syscall(libc::SYS_sigaltstack),
-                    allow_syscall(libc::SYS_write),
-                ]
-                .into_iter()
-                .collect(),
-                SeccompAction::Trap,
-            )
-            .unwrap();
-
-            assert!(SeccompFilter::apply(filter.try_into().unwrap()).is_ok());
-            assert_eq!(METRICS.seccomp.num_faults.count(), 0);
-
-            // Call the blacklisted `SYS_mkdirat`.
-            unsafe { syscall(libc::SYS_mkdirat, "/foo/bar\0") };
-
-            // Call SIGBUS signal handler.
-            assert_eq!(METRICS.signals.sigbus.count(), 0);
-            unsafe {
-                syscall(libc::SYS_kill, process::id(), SIGBUS);
-            }
+    Ok(mask)
+}
 
-            // Call SIGSEGV signal handler.
-            assert_eq!(METRICS.signals.sigsegv.count(), 0);
-            unsafe {
-                syscall(libc::SYS_kill, process::id(), SIGSEGV);
-            }
+/// Handle the signal according to its signal number.
+fn handle_signal(siginfo: signalfd_siginfo) {
```

Comment:
I was on the fence about this, since it does not need anything from the SignalManager state.
I agree however that it is cleaner to have it in the same scope

Reply:
done